### PR TITLE
Correctly parse constructor strings.

### PR DESCRIPTION
**Pull request by wanderview**

### CODE REVIEW

**Comment by wanderview** [comment]
Note, this change includes 2 changes to path-to-regexp.  These have already been incorporated and pushed to the branches at:

https://github.com/wanderview/path-to-regexp/branches

**Comment by wanderview** [comment]
This PR is another step towards fixing #7.

File: src/url-pattern-parser.ts [hunk 1]
```diff
@@ -0,0 +1,484 @@
+// The parse has been translated from the chromium c++ implementation at:
```
Comment (kenchris) [inline src/url-pattern-parser.ts:1]: C++

**Review by kenchris — APPROVED** [review]
Looks fine. Was wondering whether the "END" has to be a string or not

**Comment by wanderview** [comment]
Thanks!

> Looks fine. Was wondering whether the "END" has to be a string or not

The path-to-regexp type is not an enum, so I think it has to be a string.  We could propose changing that upstream, but I chose not to here.  I guess its a bit confusing since I made the state type an enum.

**Pull request merged**